### ISSUE TITLE
[Bug]: CommonVoice sorting is likely broken with certain torchaudio versions

### ISSUE THREAD

**Issue by AsuMagic** [speechbrain/speechbrain#1925]
### Describe the bug

CommonVoice is a dataset in MP3 format. It seems that a recent `torchaudio` version (linked issue reports `0.12`, but my `0.11` environment seems to reproduce) more or less intentionally broke querying the length of MP3 files when using `torchaudio.info(path).num_frames`, as it seems to always return 0, see <https://github.com/pytorch/audio/issues/2524>.

It seems that a PR was merged at some point to solve that issue but the minimum torchaudio version we require is probably still affected.

SpeechBrain uses this to generated the CSVs. As a result, they contain a `0.0` duration column.

`train.py` uses that column to sort data.

### Expected behaviour

`duration` should never be zero.

### To Reproduce

Run any preprocessing step in a CommonVoice recipe, then inspect `results/<seed>/save/*.csv`.

### Relevant log output

Example contents of CSV:

```
ID,duration,wav,spk_id,wrd
common_voice_fr_19642221,0.0,/users/sdelangen/CommonVoice8.0/fr/clips/common_voice_fr_19642221.mp3,4e64587f8dd935d70e50607fe60f73b5badcd6e967db2c331a6c92ca00e9d1ddde6ff2385cd1a90c6d5ceea960e708248914dcb6cdc69fe6ea7edd0208a3ff5e,C EST LE PLUS VASTE ET LE PLUS PEUPLÉ DE LA COMMUNE
common_voice_fr_19642222,0.0,/users/sdelangen/CommonVoice8.0/fr/clips/common_voice_fr_19642222.mp3,4e64587f8dd935d70e50607fe60f73b5badcd6e967db2c331a6c92ca00e9d1ddde6ff2385cd1a90c6d5ceea960e708248914dcb6cdc69fe6ea7edd0208a3ff5e,CES SEPT QUATUORS ONT ÉTÉ ÉDITÉS PAR MARC LOMBARD
common_voice_fr_19642223,0.0,/users/sdelangen/CommonVoice8.0/fr/clips/common_voice_fr_19642223.mp3,
```

**Comment by AsuMagic** [comment]
MP3 `.info` seems to be correct again with `torchaudio == 2.0.0` (and potentially earlier). However, on that version, it doesn't seem to be actually any faster than `.load(path, normalize=False)[0].size(1)`, so it's likely that it does the same thing.

So for MP3 formats, working around it in SB by doing a full load doesn't seem too unreasonable. This should not be generalized to other formats, which aren't as awful and generally have a more trivial way to check for length :)

**Comment by AsuMagic** [comment]
`DVoice` is likely also affected as it seems to use MP3 too (if that isn't an artifact from the recipe being based on CommonVoice).
